extensions/shared/external-media/media-browser/index.js

```diff
@@ -63,14 +64,22 @@ function MediaBrowser( props ) {
 		[ className ]: true,
 	} );
 
+	const onLoadMoreClick = () => {
+		if ( media.length ) {
+			setFocusedItemIndex( media.length );
+		}
+		nextPage();
+	};
+
 	return (
 		<div className={ wrapper }>
 			<div className={ classes }>
-				{ media.map( item => (
+				{ media.map( ( item, index ) => (
 					<MediaItem
 						item={ item }
 						key={ item.ID }
 						onClick={ onSelectImage }
+						isFocused={ index === focusedItemIndex }
```

Comment:
If we _do_ need to maintain this state, then I think we should rename this. The `isFocused` prop here is only valid after a Load More action. It doesn't stay up-to-date as you're moving around (which I think is unnecessary). So, if we do have to keep it as a stateful thing, let's rename this. But, I _think_ there's probably a way to do it without using the state. I bet you know more about this than I do though :)

Reply:
Good point about the prop name. Regardless if we keep it stateful or not, it would be good to name it i.e. `isInitiallyFocused` - how does that sound?

Reply:
Maybe `focusOnLoad`?

Reply:
One more condition we'd need to account for here. What if no more media items load (no more items to load, or a failure)? Will the items potentially rerender and the previously loaded image get focused? 

Do we need to build in a "only focus the item if it's different than `media.length`?

Reply:
>  Will the items potentially rerender and the previously loaded image get focused?

I tested this out by being offline, and the images don't rerender. Not sure if we should still try to account for this or not though, as it is a potential edge case.

Reply:
> What if no more media items load (no more items to load, or a failure)? Will the items potentially rerender and the previously loaded image get focused?

That wouldn't happen because we set the focus index based on **current** media items' count every time the "Load more" button is clicked. So, when a "Load more" is clicked and there are no additional items loaded, nothing will get focused because the `focusedItemIndex` would point to a non-existent (`last + 1`) item. Therefore, I don't think we need an additional check here!

Reply:
> Maybe `focusOnLoad`?

Ended up using `focusOnMount` (as we do i.e. [for the `Popover` component](https://github.com/Automattic/jetpack/blob/795f8dab3cff198b5111125a041171d2d4a5794e/extensions/blocks/map/lookup/index.js#L197)).